### PR TITLE
Updates to Modules-page

### PR DESCRIPTION
**Pull request by fflaten**
Changes:
- Migrates https://github.com/pester/Pester/wiki/Testing-different-module-types
- Adds sections/headlines to improve navigation
- Minor adjustments to text

Fix pester/pester#1468
Fix pester/pester#1367

### CODE REVIEW

**Comment by fflaten** [comment]
Waiting on pester/pester#2234 and Pester 5.4 release

File: docs/usage/modules.mdx [hunk 1]
```diff
@@ -29,9 +31,10 @@ function Get-NextVersion { return 0 }
 Export-ModuleMember -Function BuildIfChanged
 ```
 
+## Testing public functions
 You wish to write a unit test for this module which mocks the calls to `Get-Version` and `Get-NextVersion` from the module's `BuildIfChanged` command. In older versions of Pester, this was not possible. As of version 3.0, there are two ways you can perform unit tests of PowerShell script modules. The first is to inject mocks into a module:
 
-For these example, we'll assume that the PSM1 file is named "MyModule.psm1", and that it is installed on your PSModulePath.
+For these example, we'll assume the module above is installed in a path defined in you [`$env:PSModulePath`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_psmodulepath).
```
Comment (fflaten) [inline docs/usage/modules.mdx:37]:
```suggestion
For these example, we'll assume the module above is installed in a path defined in [`$env:PSModulePath`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_psmodulepath).
```

File: docs/usage/modules.mdx [hunk 2]
```diff
@@ -84,17 +85,22 @@ Describe "BuildIfChanged" {
...
+When you write your test script this way, you can mock commands that are called by the module's internal functions. However, your test script is still limited to accessing the public, exported members of the module. If you wanted to write a unit test that calls `Build` directly, for example, it wouldn't work using the above technique.
+
+## Testing private functions
+
+That's where the second approach to script module testing comes into play. With Pester's [InModuleScope](../commands/InModuleScope) command, you can cause entire sections of your test script to execute inside the targeted script module. This gives you access to non-exported members of the module. For example:
```
Comment (fflaten) [inline docs/usage/modules.mdx:96]:
```suggestion
When you write your test script this way, you can mock commands that are called by the module's internal functions. However, your test script is still limited to accessing the public, exported members of the module. If you wanted to write a unit test that calls `Build` directly, for example, it wouldn't work using the above technique. That's where the second approach to script module testing comes into play. 

## Testing private functions

With Pester's [InModuleScope](../commands/InModuleScope) command, you can cause entire sections of your test script to execute inside the targeted script module. This gives you access to non-exported members of the module. For example:
```

File: docs/usage/modules.mdx (hunk 1 above)
Comment (fflaten) [inline docs/usage/modules.mdx:37]:
```suggestion
For these example, we'll assume the module above is installed in a path defined in [`$env:PSModulePath`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_psmodulepath).
```

File: docs/usage/modules.mdx [hunk 3]
```diff
@@ -121,3 +127,166 @@ Notice that when using `InModuleScope`, you no longer need to specify a `-Module
...
+---------- -------    ----
+Script     0.0        __DynamicModule_11b8a091-bd9b-49...
+Binary     1.0.0.0    CimCmdlets
+Manifest   3.1.0.0    Microsoft.PowerShell.Management
+Manifest   3.1.0.0    Microsoft.PowerShell.Utility
+Script     5.3.3      Pester
+Script     2.0.0      PSReadline
+```
+
+To inspect your modules you might need to use `-ListAvailable` or load the module first, using `Import-Module` and then inspect it.
```
Comment (fflaten) [inline docs/usage/modules.mdx:156]:
```suggestion
To inspect your modules you might need to use `-ListAvailable` or load the module first using `Import-Module` and then inspect it.
```

File: docs/usage/modules.mdx [hunk 4]
```diff
@@ -121,3 +127,166 @@ Notice that when using `InModuleScope`, you no longer need to specify a `-Module
...
+
+:::tip
+Read more about the different module types at Microsoft Docs, see [Understanding a Windows PowerShell Module](https://docs.microsoft.com/en-us/powershell/scripting/developer/module/understanding-a-windows-powershell-module).
+:::
+
+### Usage and workarounds
+
+Pester can be used to test the behavior of commands that are exported from all types of modules. Mocking exported commands is also possible in general. For example the following test will call the `Invoke-PublicMethod` command regardless of whether it is defined in a Script, Binary, Manifest or Dynamic module:
```
Comment (fflaten) [inline docs/usage/modules.mdx:164]:
```suggestion
Pester can be used to both test and mock the behavior commands that are exported from all types of modules. For example the following tests will call the real `Invoke-PublicMethod` command and call a mocked implementation of it regardless of whether it is defined in a Script, Binary, Manifest or Dynamic module:
```

**Comment by fflaten** [comment]
Needs a thorough readthrough when you have time @nohwnd 🙂

**Comment by fflaten** [comment]
Bump

**Comment by bravo-kernel** [comment]
is this good to merge @nohwnd @fflaten

**Comment by fflaten** [comment]
Was waiting on a review, but let's merge and fix later if necessary. 🙂

**Comment by bravo-kernel** [comment]
🫡

**Pull request merged**